psutil/_psutil_osx.c

```diff
@@ -836,6 +836,51 @@ psutil_disk_partitions(PyObject *self, PyObject *args) {
 }
 
 
+static PyObject *
+psutil_disk_usage_used(PyObject *self, PyObject *args) {
+    PyObject *default_value;
+    PyObject *mount_point_bytes = NULL;
+    char* mount_point;
+
+#if PY_MAJOR_VERSION >= 3
+    if (!PyArg_ParseTuple(args, "O&O", PyUnicode_FSConverter, &mount_point_bytes, &default_value)) {
+        return NULL;
+    }
+    mount_point = PyBytes_AsString(mount_point_bytes);
+    if (NULL == mount_point) {
+        Py_XDECREF(mount_point_bytes);
+        return NULL;
+    }
+#else
+    if (!PyArg_ParseTuple(args, "sO", &mount_point, &default_value)) {
+        return NULL;
+    }
+#endif
+
+#ifdef ATTR_VOL_SPACEUSED
+    /* Call getattrlist(ATTR_VOL_SPACEUSED) to get used space info. */
+    int ret;
+    struct {
+        uint32_t size;
+        uint64_t spaceused;
+    } __attribute__((aligned(4), packed)) attrbuf = {0};
+    struct attrlist attrs = {0};
+    Py_BEGIN_ALLOW_THREADS
+    attrs.bitmapcount = ATTR_BIT_MAP_COUNT;
+    attrs.volattr = ATTR_VOL_INFO | ATTR_VOL_SPACEUSED;
+    ret = getattrlist(mount_point, &attrs, &attrbuf, sizeof(attrbuf), 0);
+    Py_END_ALLOW_THREADS
```

Comment:
2 minor suggestions:

* Py_BEGIN_ALLOW_THREADS should be moved to line 870, so that it only targets the `getattrlist` call
* one extra line after variables declaration